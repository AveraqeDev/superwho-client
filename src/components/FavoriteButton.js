import React from 'react';

import '../styles/FavoriteButton.css';

const FavoriteButton = ({ heroId }) => {

  const onClick = e => {
    e.preventDefault();
    e.currentTarget.classList.toggle('is_animating');

    e.currentTarget.classList.add('active');
  };

  const onAnimationEnd = e => {
    e.preventDefault();
    e.currentTarget.classList.toggle('is_animating');
  }
  return (
    <span 
      className={'favme'}
      onClick={onClick}
      onAnimationEnd={onAnimationEnd}
    >â™¥</span>
  );
};

export default FavoriteButton;