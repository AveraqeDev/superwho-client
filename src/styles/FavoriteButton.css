.favme {
  display: block;
  font-size: ms(2);
  cursor: pointer;
  transition: all .2s ease;
  color: #CBCDCE;
  margin: 0;
  padding: 0;
  position: relative;
}

.favme:hover {
  animation: favme-hover .3s infinite alternate;
}

.active {
  color: #DC3232;
}

.is_animating {
  animation: favme-anime .3s;
}

@keyframes favme-anime {
  0% {
    opacity: 1;
    font-size: ms(0);
    -webkit-text-stroke-color: transparent;
  }
  25% {
    opacity: .6;
    color: #ffffff;
    font-size: ms(-2);
    -webkit-text-stroke-width: 1px;
    -webkit-text-stroke-color: #DC3232;
  }
  75% {
    opacity: .6;
    color: #ffffff;
    font-size: ms(3);
    -webkit-text-stroke-width: 1px;
    -webkit-text-stroke-color: #DC3232;
  }
  100% {
    opacity: 1;
    font-size: ms(2);
    -webkit-text-stroke-color: transparent;
  }
}

@keyframes favme-hover {
  from {
    font-size: ms(3);
  }
  80% {
    font-size: ms(2);
  }
}