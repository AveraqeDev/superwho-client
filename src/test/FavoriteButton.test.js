import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FavoriteButton from '../components/FavoriteButton';

describe('<FavoriteButton />', () => {
  const testData = 1;

  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FavoriteButton heroId={testData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <FavoriteButton heroId={testData} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});