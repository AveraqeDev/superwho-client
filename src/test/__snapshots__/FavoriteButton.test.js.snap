// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteButton /> renders the UI as expected 1`] = `
<span
  className="favme"
  onAnimationEnd={[Function]}
  onClick={[Function]}
>
  â™¥
</span>
`;
